frontend/src/SignUp/SignUp.jsx

```diff
@@ -0,0 +1,113 @@
+import './SignUp.css'
+import React, { useContext, useState } from "react";
+import {useNavigate,Link} from "react-router-dom"
+import user_icon from '../assets/person.png'
+import email_icon from '../assets/email.png'
+import password_icon from '../assets/password.png'
+
+
+function SignUp(){
+    const navigate = useNavigate()
+    const [name, setName] = useState('')
+    const [email, setEmail] = useState('')
+    const [password, setPassword] = useState('')
+    const [confirmPassword, setConfirmPassword] = useState('')
+    const {updateUser} = useContext(UserContext)
+
+
+    async function handleSignUp(event){
+        event.preventDefault();
+        try{
+            const response = await fetch("http://localhost:2500/logIn",{
+                method:"POST",
+                headers:{
+                    "Content-Type": "application/json",
+                },
+                body:Json.stringify({
+                    name,
+                    email,
+                    password,
+                    confirmPassword,
+                    role,
+                }),
+                credentials: 'include'
+            });
+            const data = await response.json()
+            if(response.ok){
+                const loggedInUser = data.user
+                updateUser(loggendInUser)
+                navigate('/projects')
+            } else{
+                setError('Login failed')
```

Comment:
You are using this string in another file as well - let's create a file just for string and move all the strings that are used in multiple files into there